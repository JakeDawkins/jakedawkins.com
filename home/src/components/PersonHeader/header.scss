@import "../../styles/variables";

//@extend-elements
//original selectors
//.header-site .content, .header-post .content
%extend_1 {
  width: 95%;
  position: absolute;
  //Instead of the line below you could use @include transform-style($style)
  transform-style: preserve-3d;
  top: 50%;
  left: 50%;
  //Instead of the line below you could use @include transform($scale, $rotate, $transx, $transy, $skewx, $skewy, $originx, $originy)
  transform: translate(-50%,-50%);
  text-align: center;
}

.header-site {
  background: $color--blue;
  height: 100%;
  .content {
    @extend %extend_1;
  }
  h1 {
    color: $color--white;
    text-align: center;
  }
  h2 {
    color: $color--white;
    text-align: center;
  }
  .site-title {
    font-size: 3.75rem;
    line-height: 1;
    margin-top: -80px;
    display: block;
  }
  .site-description {
    font-size: $font-size--h4;
    display: block;
    margin: 1.875rem 0;
  }
}
.header-post {
  background: $color--blue;
  height: 100%;
  .content {
    @extend %extend_1;
  }
  .subheader {
    display: table;
    padding: 20px;
    a {
      border: 2px solid $color--white;
      //Instead of the line below you could use @include border-radius($radius, $vertical-radius)
      border-radius: 20px;
      padding: 0 10px;
    }
  }
  .date {
    font-family: $font--header;
    font-weight: $weight--normal;
    font-style: normal;
    font-size: 1rem;
    color: $color--white;
    text-align: center;
  }
  h1 {
    text-transform: uppercase;
    font-size: 1.875rem;
    margin: 0 0 30px;
    color: $color--white;
    text-align: center;
  }
  .subtitle {
    font-family: $font--header;
    font-weight: $weight--normal;
    font-style: normal;
    font-size: 1.25rem;
    color: $color--white;
    text-align: center;
  }
  p {
    color: $color--white;
    text-decoration: none;
    font-family: $font--header;
    font-weight: $weight--normal;
    font-style: normal;
    font-size: 1.125rem;
  }
}
@media only screen and(min-width:37.5rem) {
  .header-site {
    .site-title {
      margin-bottom: 0;
      font-size: 6.25rem;
    }
    .site-description {
      font-size: $font-size--h4;
    }
  }
  .header-post {
    .content {
      *zoom: 1;
      width: auto;
      max-width: 1000px;
      float: none;
      display: block;
      margin-right: auto;
      margin-left: auto;
      padding-left: 0;
      padding-right: 0;
      &:before {
        content: '';
        display: table;
      }
      &:after {
        content: '';
        display: table;
        clear: both;
      }
    }
    h1 {
      font-size: 3.75rem;
    }
    .subheader a {
      padding: 0 10px;
    }
  }
}
