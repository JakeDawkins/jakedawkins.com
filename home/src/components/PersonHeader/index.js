import React, { PropTypes } from "react";
import "./header.scss";
import { Link } from "react-router";

import MenuWithSearch from "../MenuWithSearch";

const PersonHeader = (props) => (
  <header className="header-site" role="banner">
    <div className="content">
      <h1>
        <span className="site-title">{props.userDisplayName}</span>
        <span className="site-description">{props.userDescription}</span>
      </h1>
      <div className="icons-home">
        <a aria-label="Send email" href={`mailto:${props.userEmail}`}>
          <svg className="icon icon-email">
            <use xlinkHref="#icon-email"></use>
          </svg>
        </a>
        <a aria-label="My Twitter" target="_blank" href={`https://twitter.com/${props.userTwitter}`}>
          <svg className="icon icon-twitter">
            <use xlinkHref="#icon-twitter"></use>
          </svg>
        </a>
        <a aria-label="My Github" target="_blank" href={`https://github.com/${props.userGithub}`} >
          <svg className="icon icon-github-alt">
            <use xlinkHref="#icon-github-alt"></use>
          </svg>
        </a>
      </div>
      <div>
        <Link to="/about?id=1" style={{textDecoration: "none"}}>
          <br /><br />
          <h2>About Me</h2>
        </Link>
      </div>
    </div>
    <a role="button" className="down" data-scroll href="#scroll"><svg className="icon icon-angle-down"><use xlinkHref="#icon-angle-down"></use></svg></a>
    <MenuWithSearch />
  </header>
);

PersonHeader.propTypes = {
  userDisplayName: PropTypes.string,
  userDescription: PropTypes.string,
  userEmail: PropTypes.string,
  userTwitter: PropTypes.string,
  userGithub: PropTypes.string,
};

export default PersonHeader;
