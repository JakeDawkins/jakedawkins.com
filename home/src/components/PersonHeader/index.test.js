import PersonHeader from './';
import React from 'react';
import renderer from 'react-test-renderer';

describe("PersonHeader", () => {

  it("should exist", () => {
    const personHeader = renderer.create(
      <PersonHeader />
    );

    expect(personHeader).toMatchSnapshot();
  });

  it("should render props correctly", () => {
    // const personHeader = renderer.create(
    //   <PersonHeader />
    // );

    const personHeaderWithProps = renderer.create(
      <PersonHeader
        userDisplayName="test display name"
        userDescription="test description"
        userEmail="test@test.com"
        userTwitter="tw"
        userGithub="gh"
      />
    );

    expect(personHeaderWithProps).toMatchSnapshot();
    // expect(personHeaderWithProps).not.toBe(personHeader);
  });

});
