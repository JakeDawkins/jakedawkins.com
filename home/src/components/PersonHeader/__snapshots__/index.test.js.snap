exports[`PersonHeader should exist 1`] = `
<header
  className="header-site"
  role="banner">
  <div
    className="content">
    <h1>
      <span
        className="site-title" />
      <span
        className="site-description" />
    </h1>
    <div
      className="icons-home">
      <a
        aria-label="Send email"
        href="mailto:undefined">
        <svg
          className="icon icon-email">
          <use
            xlinkHref="#icon-email" />
        </svg>
      </a>
      <a
        aria-label="My Twitter"
        href="https://twitter.com/undefined"
        target="_blank">
        <svg
          className="icon icon-twitter">
          <use
            xlinkHref="#icon-twitter" />
        </svg>
      </a>
      <a
        aria-label="My Github"
        href="https://github.com/undefined"
        target="_blank">
        <svg
          className="icon icon-github-alt">
          <use
            xlinkHref="#icon-github-alt" />
        </svg>
      </a>
    </div>
    <div>
      <a
        onClick={[Function bound handleClick]}
        style={
          Object {
            "textDecoration": "none"
          }
        }>
        <br />
        <br />
        <h2>
          About Me
        </h2>
      </a>
    </div>
  </div>
  <a
    className="down"
    data-scroll={true}
    href="#scroll"
    role="button">
    <svg
      className="icon icon-angle-down">
      <use
        xlinkHref="#icon-angle-down" />
    </svg>
  </a>
  <div>
    <div
      className="search-wrapper">
      <div
        className="search-form">
        <input
          className="search-field"
          placeholder="Search..."
          type="text" />
        <svg
          className="icon-remove-sign">
          <use
            xlinkHref="#icon-close" />
        </svg>
        <ul
          className="search-results search-list" />
      </div>
    </div>
    <div
      className="overlay"
      id="fade"
      onClick={[Function bound toggleMenu]} />
    <a
      className="slideButton fade"
      id="slide">
      <svg
        className="icon-menu"
        id="open"
        onClick={[Function bound toggleMenu]}
        style={
          Object {
            "display": "inline"
          }
        }>
        <use
          xlinkHref="#icon-menu" />
      </svg>
      <svg
        className="icon-menu"
        id="close"
        onClick={[Function bound toggleMenu]}
        style={
          Object {
            "display": "none"
          }
        }>
        <use
          xlinkHref="#icon-close" />
      </svg>
    </a>
    <aside
      className=""
      id="sidebar">
      <nav
        id="navigation">
        <h2>
          MENU
        </h2>
        <ul>
          <li>
            <a
              onClick={[Function bound handleClick]}
              style={Object {}}>
              Posts
            </a>
          </li>
          <li>
            <a
              onClick={[Function bound handleClick]}
              style={Object {}}>
              About
            </a>
          </li>
        </ul>
      </nav>
    </aside>
  </div>
</header>
`;

exports[`PersonHeader should render props correctly 1`] = `
<header
  className="header-site"
  role="banner">
  <div
    className="content">
    <h1>
      <span
        className="site-title">
        test display name
      </span>
      <span
        className="site-description">
        test description
      </span>
    </h1>
    <div
      className="icons-home">
      <a
        aria-label="Send email"
        href="mailto:test@test.com">
        <svg
          className="icon icon-email">
          <use
            xlinkHref="#icon-email" />
        </svg>
      </a>
      <a
        aria-label="My Twitter"
        href="https://twitter.com/tw"
        target="_blank">
        <svg
          className="icon icon-twitter">
          <use
            xlinkHref="#icon-twitter" />
        </svg>
      </a>
      <a
        aria-label="My Github"
        href="https://github.com/gh"
        target="_blank">
        <svg
          className="icon icon-github-alt">
          <use
            xlinkHref="#icon-github-alt" />
        </svg>
      </a>
    </div>
    <div>
      <a
        onClick={[Function bound handleClick]}
        style={
          Object {
            "textDecoration": "none"
          }
        }>
        <br />
        <br />
        <h2>
          About Me
        </h2>
      </a>
    </div>
  </div>
  <a
    className="down"
    data-scroll={true}
    href="#scroll"
    role="button">
    <svg
      className="icon icon-angle-down">
      <use
        xlinkHref="#icon-angle-down" />
    </svg>
  </a>
  <div>
    <div
      className="search-wrapper">
      <div
        className="search-form">
        <input
          className="search-field"
          placeholder="Search..."
          type="text" />
        <svg
          className="icon-remove-sign">
          <use
            xlinkHref="#icon-close" />
        </svg>
        <ul
          className="search-results search-list" />
      </div>
    </div>
    <div
      className="overlay"
      id="fade"
      onClick={[Function bound toggleMenu]} />
    <a
      className="slideButton fade"
      id="slide">
      <svg
        className="icon-menu"
        id="open"
        onClick={[Function bound toggleMenu]}
        style={
          Object {
            "display": "inline"
          }
        }>
        <use
          xlinkHref="#icon-menu" />
      </svg>
      <svg
        className="icon-menu"
        id="close"
        onClick={[Function bound toggleMenu]}
        style={
          Object {
            "display": "none"
          }
        }>
        <use
          xlinkHref="#icon-close" />
      </svg>
    </a>
    <aside
      className=""
      id="sidebar">
      <nav
        id="navigation">
        <h2>
          MENU
        </h2>
        <ul>
          <li>
            <a
              onClick={[Function bound handleClick]}
              style={Object {}}>
              Posts
            </a>
          </li>
          <li>
            <a
              onClick={[Function bound handleClick]}
              style={Object {}}>
              About
            </a>
          </li>
        </ul>
      </nav>
    </aside>
  </div>
</header>
`;
